deploy/helm/sumologic/templates/traces/otel-gateway/deployment.yaml

```diff
@@ -0,0 +1,113 @@
+{{- if and (eq .Values.sumologic.traces.enabled true) (eq .Values.otelgateway.enabled true) }}
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{ template "sumologic.metadata.name.otelgateway.deployment" . }}
+  labels:
+    app: {{ template "sumologic.labels.app.otelgateway.pod" . }}
+    component: {{ template "sumologic.labels.app.otelgateway.component" . }}
+    {{- include "sumologic.labels.common" . | nindent 4 }}
+spec:
+  minReadySeconds: 5
+  selector:
+    matchLabels:
+      app: {{ template "sumologic.labels.app.otelgateway.pod" . }}
+      component: {{ template "sumologic.labels.app.otelgateway.component" . }}
+{{- if .Values.sumologic.podLabels }}
+{{ toYaml .Values.sumologic.podLabels | indent 6 }}
+{{- end }}
+{{- if .Values.otelgateway.deployment.podLabels }}
+{{ toYaml .Values.otelgateway.deployment.podLabels | indent 6 }}
+{{- end }}
+  template:
+    metadata:
+      annotations:
+{{- if .Values.sumologic.podAnnotations }}
+{{ toYaml .Values.sumologic.podAnnotations | indent 8 }}
+{{- end }}
+{{- if .Values.otelgateway.deployment.podAnnotations }}
+{{ toYaml .Values.otelgateway.deployment.podAnnotations | indent 8 }}
+{{- end }}
+      labels:
+        app: {{ template "sumologic.labels.app.otelgateway.pod" . }}
+        component: {{ template "sumologic.labels.app.otelgateway.component" . }}
+        {{- include "sumologic.labels.common" . | nindent 8 }}
+{{- if .Values.sumologic.podLabels }}
+{{ toYaml .Values.sumologic.podLabels | indent 8 }}
+{{- end }}
+{{- if .Values.otelgateway.deployment.podLabels }}
+{{ toYaml .Values.otelgateway.deployment.podLabels | indent 6 }}
+{{- end }}
+    spec:
+      serviceAccountName: {{ template "sumologic.metadata.name.roles.serviceaccount" . }}
+      {{- if .Values.otelgateway.deployment.nodeSelector }}
+      nodeSelector:
+        {{ toYaml .Values.otelgateway.deployment.nodeSelector | indent 8 }}
+      {{- end }}
+      {{- if .Values.otelgateway.deployment.tolerations }}
+      tolerations:
+        {{ toYaml .Values.otelgateway.deployment.tolerations | indent 8 }}
+      {{- end }}
+      {{- if .Values.otelgateway.deployment.priorityClassName }}
+      priorityClassName: {{ .Values.otelgateway.deployment.priorityClassName | quote }}
+      {{- end }}
+      # Otel agent quits if the load balancing backend (collector/samples) is
+      # not ready during first connect attempt. Restart policy `Always` guarantees that
+      # the agent will have a chance to retry when collector is ready.
+      restartPolicy: Always
+      containers:
+      - name: otelcontrib-gateway
+        image: {{ .Values.otelgateway.deployment.image.repository }}:{{ .Values.otelgateway.deployment.image.tag }}
+        imagePullPolicy: {{ .Values.otelgateway.deployment.image.pullPolicy }}
+        command:
+          - "/otelcontribcol"
+          - "--config=/conf/traces.otelgateway.conf.yaml"
+        env:
+        - name: GOGC
+          value: "80"
```

Comment:
Why are we changing GC behaviour?

Reply:
It's a copy from agent config.

Reply:
We can leave it as is, but I suspect what the author really wanted was a good setting for memory ballast.